### PR TITLE
Updated ClusterController with the relationship between cluster and AZ

### PR DESCRIPTION
**Pull request by kimeunju108**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:
This PR proposes to resolve the following three things:
- one-to-one relationship between cluster and AvailabilityZone. 
- sample cluster dataset with new AZ data
- gRPC cluster delete function

**Which issue(s) this PR fixes**:

Fixes #213 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: 
NONE

### CODE REVIEW

File: globalscheduler/test/yaml/sample_1000_clusters.yaml [hunk 1]
```diff
@@ -28891,7 +28891,7 @@ spec:
 apiVersion: globalscheduler.com/v1
 kind: Cluster
 metadata:
-  name: SE-2|az100
+  name: se-2.az100
```
Comment (cathyhongzhang) [inline globalscheduler/test/yaml/sample_1000_clusters.yaml]: don't you need to specify geolocation for each cluster?

Comment (kimeunju108) [reply globalscheduler/test/yaml/sample_1000_clusters.yaml]: Kubernetes allow only lower alphanumeric, -, and . for name

File: globalscheduler/controllers/cluster/cluster_controller.go [hunk 2]
```diff
@@ -326,36 +341,49 @@ func (c *ClusterController) gRPCRequest(event EventType, clusterNameSpace string
...
+	ipAddress := cluster.Spec.IpAddress
+	region := cluster.Spec.Region.Region
+	az := cluster.Spec.Region.AvailabilityZone
+	siteID := fmt.Sprintf("%s|%s", region, az)
+	clusterName := cluster.Name
+	if ipAddress == nil || region == nil || az == nil || clusterName == nil  {
+		err = fmt.Errorf("required cluster spec data is null")
+		return verified
+	}
+	verified = (clusterName == siteID)
```
Comment (cathyhongzhang) [inline globalscheduler/controllers/cluster/cluster_controller.go]: you do not need to verify ckusterName==siteID

**Review by cathyhongzhang — APPROVED** [review]

**Comment by cathyhongzhang** [comment]
Eunju will update this PR after merging of PR 229.

**Review by CoderKevinZhang — APPROVED** [review]

**Pull request merged**